### PR TITLE
Some improvement

### PR DESCRIPTION
**Pull request by q792602257**
press esc to exit the parameters View Dialog
ReturnValue filter && Parameter Filter && regexp Match( need startswith <code>/</code>)
Add a simple PHPValue Parser
Make the inspector Dialog readonly and press <code>F</code> can prettify the phpValue

### CODE REVIEW

**Comment by q792602257** [comment]
there is still have a bug;
sometimes the times is negative(<0)

File: XtRay.Windows/MainWindow.xaml.cs [hunk 1]
```diff
@@ -19,12 +21,21 @@ public partial class MainWindow : Window
...
+
+        #endregion
 
         public MainWindow()
         {
             InitializeComponent();
-            var textChanged = ((EventHandler<TextChangedEventArgs>)SearchBox_TextChanged).Debounce(444);
+            var textChanged = ((EventHandler<EventArgs>)ApplyFilterEvent).Debounce(444);
             SearchBox.TextChanged += (s, e) => textChanged(s, e);
+            ComboBox.SelectionChanged += (s, e) => textChanged(s, e);
```
Comment (gasparportik) [inline XtRay.Windows/MainWindow.xaml.cs]: Move this line after the line where you set the ItemsSource(`ComboBox.ItemsSource = ParserList;`) and the SelectedIndex, that way the event won't fire programmatically.

**Comment by gasparportik** [comment]
Hey @q792602257,
why did you close this PR? Things look good and I like the features you added.
Would you be willing to finalize the changes?

**Comment by q792602257** [comment]
I think that issue made by me then I close the PR,
and I realize I can't fix this bug TAT

**Comment by gasparportik** [comment]
No no, the negative time bug was probably my fault, the only problem I've seen with your code was with the filter selector, and I've made a comment to the code on how you can fix that.
Also I'm unfamiliar with TAT, please tell me what it means. :)

**Comment by q792602257** [comment]
TAT means 😭, 'A' is present the mouth 'T' is the eye with tear

**Comment by gasparportik** [comment]
o_o, ^.^, TAT, yeah now I see the similarity.

**Pull request merged**